Kudu.Services/Infrastructure/UriHelper.cs

```diff
@@ -1,9 +1,29 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
 
 namespace Kudu.Services.Infrastructure
 {
     public static class UriHelper
     {
+        public static Uri GetBaseUri(HttpRequestMessage request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException("request");
+            }
+
+            IEnumerable<string> disguisedHostValues = new List<string>();
+            if (request.Headers.TryGetValues("DISGUISED-HOST", out disguisedHostValues) && disguisedHostValues.Count() > 0)
```

Comment:
FrontedEnd will always pass this header to Kudu (confirmed with Jenny), and it always carry the right host name.
 
when running Kudu on mono in a container, container is bind to a local ip, so we cannot rely on "request.RequestUri", which will be "127.0.0.1:xxxx". hence this fix. i will update code to have more comments.